### PR TITLE
Allow `throw null`

### PR DESCRIPTION
**Pull request by alxarch**
Throwing `null` results in a `TypeError` due to a property check in `helpers.updateSyntaxError` and an `err.stack` access in `command.compileScript()`

### CODE REVIEW

File: test/helpers.coffee [hunk 1]
```diff
@@ -123,3 +123,11 @@ test "the `baseFileName` helper returns the file name to write to", ->
     name = baseFileName sourceFileName, yes
     filename = name + ext
     eq filename, expectedFileName
+
+test "the `updateSyntaxError` helper allows `throw null`", ->
+    try
+        error = updateSyntaxError null
+    catch err
+        error = err
+        console.error err.stack
```
Comment (michaelficarra) [inline test/helpers.coffee]: What is this?

**Comment by michaelficarra** [comment]
Please squash your commits.

File: test/helpers.coffee (hunk 1 above)
Comment (alxarch) [inline test/helpers.coffee]: Just added a test to ensure the updateSyntaxHelper does not block null throwing in the future

**Comment by alxarch** [comment]
@michaelficarra I squashed the commits as you requested

**Comment by alxarch** [comment]
@michaelficarra I was thinking of a more generic test executing `throw null` and testing the coffeescript interpreter response. Is there a place for such tests somewhere?

File: test/helpers.coffee [hunk 2]
```diff
@@ -123,3 +123,10 @@ test "the `baseFileName` helper returns the file name to write to", ->
     name = baseFileName sourceFileName, yes
     filename = name + ext
     eq filename, expectedFileName
+
+test "the `updateSyntaxError` helper allows `throw null`", ->
```
Comment (lydell) [inline test/helpers.coffee]: Couldn’t this whole test be replaced by `eq null, updateSyntaxError null`?

Comment (alxarch) [inline test/helpers.coffee]: Without the try/catch the previous code would throw a `TypeError` that would not be caught in a tidy fashion from the test. If this is not considered an issue I can change it. I think a better place for testing this fix is `test/exception_handling.coffee` with a simple `throws (->throw null), null`. Should I proceed thus?

**Comment by vendethiel** [comment]
LGTM. @lydell @michaelficarra ?

File: test/exception_handling.coffee [hunk 3]
```diff
@@ -140,3 +140,10 @@ test "parameter-less catch clause", ->
   try throw new Error 'failed' catch finally ok true
 
   ok try throw new Error 'failed' catch then true
+
+test "Allow throwing `null`", ->
+	try
```
Comment (michaelficarra) [inline test/exception_handling.coffee]: This indentation is off.

File: test/exception_handling.coffee [hunk 4]
```diff
@@ -140,3 +140,10 @@ test "parameter-less catch clause", ->
   try throw new Error 'failed' catch finally ok true
 
   ok try throw new Error 'failed' catch then true
+
+test "Allow throwing `null`", ->
```
Comment (michaelficarra) [inline test/exception_handling.coffee]: Would this test have failed without your changes? Did you try it?
